pubsub/drivertest/drivertest.go

```diff
@@ -185,35 +188,91 @@ func testSendReceive(t *testing.T, newHarness HarnessMaker) {
 	}
 	defer cleanup()
 
-	// Send to the topic.
-	var want []*pubsub.Message
-	for i := 0; i < 3; i++ {
+	want := publishN(t, ctx, top, 3)
+	got := receiveN(t, ctx, sub, len(want))
+
+	// Check that the received messages match the sent ones.
+	if diff := diffMessageSets(got, want); diff != "" {
+		t.Error(diff)
+	}
+}
+
+// Receive from two subscriptions to the same topic.
+// Verify both get all the messages.
+func testSendReceiveTwo(t *testing.T, newHarness HarnessMaker) {
+	// Set up.
+	ctx := context.Background()
+	h, err := newHarness(ctx, t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer h.Close()
+
+	dt, err := h.MakeTopic(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	top := pubsub.NewTopic(dt)
+	defer top.Shutdown(ctx)
+
+	makeSub := func(i int) *pubsub.Subscription {
+		ds, err := h.MakeSubscription(ctx, dt, i)
+		if err != nil {
+			t.Fatal(err)
+		}
+		return pubsub.NewSubscription(ds)
+	}
+	s0 := makeSub(0)
+	s1 := makeSub(1)
+	defer s0.Shutdown(ctx)
+	defer s1.Shutdown(ctx)
+
+	want := publishN(t, ctx, top, 3)
+	gotc := make(chan []*pubsub.Message, 2)
+	go func() { gotc <- receiveN(t, ctx, s0, len(want)) }()
```

Comment:
I don't think it's necessary to have these two goroutines here. It's probably clearer to write it like this:
```go
ss := []*pubsub.Subscription{makeSub(0), makeSub(1)}
...
for _, s := range ss {
    got := receiveN(t, ctx, s, len(want))
    if diff := diffMessageSets(got, want); diff != "" {
        t.Error(diff)
    }
}
```
though maybe a little slower while recording.

Reply:
Done.